### DISCUSSION TITLE
Bazel build error

### DISCUSSION THREAD

**Discussion by kmanikanta335** [oppia/oppia-android#5216] [Q&A (Installation)]
### Operating System

windows 11

### Android Studio Version

Android Studio Flamingo | 2022.2.1 Patch 1

### RAM size

8 gb

### Free disk space

240 gb

### Which step are you stuck on?

I am new contributor to this organization and I have submitted my individual CLA survey , but have not recognized by any mentors from past 1 month .
Although I have started by journey in setup the project, but have encountered this error while building the project in WSL using BAZEL
the error was encountered while following this [path](https://github.com/oppia/oppia-android/wiki/Bazel-Setup-Instructions-for-Windows#6-verifying-the-build)
command used : bazel build //:oppia

ERROR occured
![image](https://github.com/oppia/oppia-android/assets/118070186/281da7b6-ad02-40b6-8071-727db17e1642)

### Error Log

```bash
ERROR: /home/mani114/opensource/oppia-android/app/src/main/java/org/oppia/android/app/application/dev/BUILD.bazel:12:19: //app/src/main/java/org/oppia/android/app/application/dev:developer_application_kt depends on @kt_java_stub_template//file:file in repository @kt_java_stub_template which failed to fetch. no such package '@kt_java_stub_template//file': java.io.IOException: Error downloading [https://raw.githubusercontent.com/bazelbuild/bazel/3.7.0/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt] to /home/mani114/.cache/bazel/_bazel_mani114/448f11c285b7b190ad4fd87f081da82c/external/kt_java_stub_template/file/downloaded: connect timed out
ERROR: Analysis of target '//:oppia' failed; build aborted: Analysis failed
INFO: Elapsed time: 57.807s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (3 packages loaded, 21 targets configured)
```

### Approaches already used to resolve the issue

the [URL](https://[raw.githubusercontent.com](https://raw.githubusercontent.com/bazelbuild/bazel/3.7.0/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt)/bazelbuild/bazel/3.7.0/src/main/java/com/google/devtools/build/lib/bazel/rules/java/java_stub_template.txt) its trying to reach for getting dependencies is not getting loaded

**Comment by MohitGupta121** [comment]
@kmanikanta335 We really appreciate that you filled CLA and like to contribution on oppia. 

Basically this error you getting while build is due to your Internet connection, If you are from india then there is some url blocking issue with [indian Jio network](https://timesofindia.indiatimes.com/gadgets-news/github-content-domain-blocked-for-these-indian-users-reports/articleshow/96687992.cms), So I recommend you to please try to change your network or try using VPN. Thanks!

**Reply by kmanikanta335** [reply]
Thank you sir
my question is will Oppia Android participate in GSOC'24?

**Reply by MohitGupta121** [reply]
@kmanikanta335 yes hope do, we are planning.

**Reply by kmanikanta335** [reply]
After that I encountered another error while bazel build
command used: bazel build //:oppia

ERROR
![image](https://github.com/oppia/oppia-android/assets/118070186/f7188e8e-012b-4f97-95f2-a0438a1a0d9f)


error log

ERROR: /home/mani114/opensource/oppia-android/utility/src/main/java/org/oppia/android/util/math/BUILD.bazel:101:19: KotlinCompile //utility/src/main/java/org/oppia/android/util/math:comparator_extensions_kt { kt: 1, java: 0, srcjars: 0 } for armeabi-v7a failed: Worker process did not return a WorkResponse:

---8<---8<--- Start of log, file at /home/mani114/.cache/bazel/_bazel_mani114/448f11c285b7b190ad4fd87f081da82c/bazel-workers/worker-10-KotlinCompile.log ---8<---8<---
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by io.bazel.kotlin.builder.toolchain.KotlinToolchain$JDepsInvoker (file:/home/mani114/.cache/bazel/_bazel_mani114/448f11c285b7b190ad4fd87f081da82c/external/io_bazel_rules_kotlin/src/main/kotlin/kotlin_worker.jar) to method com.sun.tools.jdeps.Main.run(java.lang.String[],java.io.PrintWriter)
WARNING: Please consider reporting this to the maintainers of io.bazel.kotlin.builder.toolchain.KotlinToolchain$JDepsInvoker
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
---8<---8<--- End of log ---8<---8<---
Target //:oppia failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 122.003s, Critical Path: 60.89s
INFO: 214 processes: 3 internal, 202 linux-sandbox, 9 worker.
FAILED: Build did NOT complete successfully
